Assets/Tests/InputSystem/CorePerformanceTests.cs

```diff
@@ -1100,4 +1102,106 @@ public void Performance_OptimizedControls_ReadingPose4kTimes(OptimizationTestTyp
     }
 
 #endif
+
+    #if UNITY_2022_3_OR_NEWER
+    [PrebuildSetup(typeof(ProjectWideActionsBuildSetup))]
+    [PostBuildCleanup(typeof(ProjectWideActionsBuildSetup))]
+    [UnityTest, Performance]
+    [Category("Performance")]
+    public IEnumerator Performance_MeasureInputSystemFrameTimeWithProfilerMarkers_FPS()
+    {
+        string[] markers =
+        {
+            "InputUpdate",
+            "InputSystem.onBeforeUpdate",
+            "InputSystem.onAfterUpdate",
+            "PreUpdate.NewInputUpdate",
+            "PreUpdate.InputForUIUpdate",
+            "FixedUpdate.NewInputFixedUpdate"
+        };
+
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+        var mouse = InputSystem.AddDevice<Mouse>();
+
+        var moveAction = InputSystem.actions.FindAction("Move");
+        var lookAction = InputSystem.actions.FindAction("Look");
+        var attackAction = InputSystem.actions.FindAction("Attack");
+        var jumpAction = InputSystem.actions.FindAction("Jump");
+        var sprintAction = InputSystem.actions.FindAction("Sprint");
+
+        int performedCallCount = 0;
+
+        moveAction.performed += context => {
+            performedCallCount++;
+        };
+
+        lookAction.performed += context => {
+            performedCallCount++;
+        };
+
+        attackAction.performed += context => {
+            performedCallCount++;
+        };
+
+        jumpAction.performed += context => {
+            performedCallCount++;
+        };
+
+        sprintAction.performed += context => {
+            performedCallCount++;
+        };
+
+        using (Measure.ProfilerMarkers(markers))
+        {
+            for (int i = 0; i < 500; ++i)
+            {
+                PressAndRelease(keyboard.wKey, queueEventOnly: true);
```

Comment:
On the other hand (compared to comment below), pressing W, A, S and D as well as shift and space and mouse button at a constant rate of e.g. 60 times per second is not very realistic? I would suggest to use a modulo operation here with a switch to do different input for each frame depending on a modulo computation, e.g. if (i % 60 == 0) PressSomething() 
would mean I press something every 60th frame, which means once a second if playing at 60 Hz. If benchmark is targeting something specific (microbenchmark) it may make sense to stick to large data amount though.

Reply:
https://github.com/Unity-Technologies/InputSystem/pull/2027/commits/79525e045620aef6636af2f751375e40b02f6862